### PR TITLE
feat(backend): forget password endpoint

### PR DESCRIPTION
**Pull request by rhizu05**
﻿## Description
Menambahkan endpoint forget password di backend beserta validasi dan pengujian.

## Related Issue
- Menambahkan handler forgetPassword di auth.controller.js.
- Endpoint memanggil authService.forgetPassword(email).
- Response dibuat 204 No Content untuk semua request valid (baik email terdaftar maupun tidak) demi keamanan.
- Error dari service diteruskan ke middleware error handler dengan next(error).
- Menambahkan/memperbarui test untuk skenario sukses, validasi, dan edge case (database/email error).

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] Refactor
- [ ] Documentation
- [ ] DevOps / CI/CD
- [ ] AI / Model update

## Area Affected
- [ ] Frontend
- [x] Backend
- [ ] AI / Data Science
- [ ] Database
- [ ] Infrastructure

## Checklist
- [x] Code follows project style guidelines
- [x] Self-review completed
- [x] Tests added/updated
- [ ] Documentation updated (if needed)
- [x] No breaking changes (or documented)

### CODE REVIEW

File: backend/src/modules/auth/auth.validation.js
Comment (ahmaadn) [inline backend/src/modules/auth/auth.validation.js:1]: pake package zod, jangan pakai package joi

File: backend/src/app.js [hunk 1]
```diff
@@ -21,11 +21,12 @@ const createApp = () => {
     res.json({ status: 'ok', service: 'e-wallet-sentiment-backend' });
   });
 
-  // Review routes
-  app.use('/api/reviews', reviewRouter);
-
   // Auth routes
   app.use('/api/auth', authRouter);
+  app.use('/auth', authRouter);
```
Comment (ahmaadn) [inline backend/src/app.js]: duplicate route, pake yang /api/auth

File: backend/package.json [hunk 2]
```diff
@@ -16,21 +16,18 @@
   },
   "dependencies": {
     "@prisma/client": "^7.4.1",
-    "bcryptjs": "^3.0.3",
     "cors": "^2.8.6",
     "dotenv": "^16.6.1",
     "express": "^4.22.1",
     "helmet": "^8.1.0",
-    "jsonwebtoken": "^9.0.3",
+    "joi": "^18.0.2",
```
Comment (ahmaadn) [inline backend/package.json]: joi hapus sajaa, saya sudah install package zod

File: backend/src/middlewares/validation.middleware.js
Comment (ahmaadn) [inline backend/src/middlewares/validation.middleware.js:1]: hapus middleware, validasi langsung di [auth.controller.js](https://github.com/Nthoel/e-wallet-sentiment/pull/23/changes#diff-4377a41e20c6a51eda8b1da690c7dae332127efbdc7920bb66610fe2a897d35b)

**Review by ahmaadn — CHANGES_REQUESTED** [review]
Revisi

**Review by ahmaadn — APPROVED** [review]
